src/main/java/io/dgraph/DgraphClient.java

```diff
@@ -31,6 +41,154 @@ public class DgraphClient {
 
   private final DgraphAsyncClient asyncClient;
 
+  /**
+   * Creates a new DgraphClient instance from a connection string.
+   *
+   * <p>This method attempts to authenticate via Dgraph's ACL mechanism if
+   * username and password are provided.
+   * <p>The connection string has the format: "dgraph://[username:password@]host:port[?params]"
+   * <p>Supported query parameters:
+   * <ul>
+   *   <li>sslmode - SSL connection mode. Supported values:
+   *     <ul>
+   *       <li>"disable" - No encryption, uses plaintext</li>
+   *       <li>"require" - Uses TLS encryption without certificate verification</li>
+   *       <li>"verify-ca" - Uses TLS encryption with certificate verification</li>
+   *     </ul>
+   *   </li>
+   *   <li>apikey - API key for authorization from Dgraph Cloud</li>
+   *   <li>bearertoken - Bearer token for authorization</li>
+   * </ul>
+   *
+   * @param connectionString The connection string to connect to Dgraph
+   * @return A new DgraphClient instance
+   * @throws IllegalArgumentException If the connection string is invalid
+   * @throws MalformedURLException If the connection string cannot be parsed as a URL
+   * @throws SSLException If there's an error configuring the SSL context for sslmode=require
+   */
+  public static DgraphClient openConnection(String connectionString)
```

Comment:
Why not just open?

Reply:
Following the naming convention of the Java postgres client.

Reply:
Link please?

Reply:
Whoops, the DriverManager's method is `getConnection()`. Maybe I'll just change this to `open()` to keep it consistent with the other clients.